### ISSUE TITLE
Freestore allocation policies and log policies

### ISSUE THREAD

**Issue by ambroise-leclerc** [ambroise-leclerc/ETL#6]
Add a std::clog log policy to freestore : 
- in debug_policies : FreeStoreNoDebug, FreeStoreDebugTrace, FreeStoreDebugClog

Add a malloc allocation policy
- FreestoreOnHeap : relies on malloc/free for C allocation compatibility since freestore and C's Heap reside in the same memory space and can overlap.
- FreestoreNative : consumes less memory space and return information about memory consumption and fragmentation.

**Comment by ambroise-leclerc** [comment]
Closing to rewrite narrowed to the still-missing scope — the log-policy half (`FreeStoreTracePolicy`/`FreeStoreDebugTrace`, gated by `ETL_FREESTORE_LOG_DEPTH`) is already implemented in `include/etl/debug_policies.h`. See the new issue for the remaining malloc-backed allocation policy.
